migrations/20220516035046_template-campaigns.js

```diff
@@ -0,0 +1,318 @@
+exports.up = function up(knex) {
+  return knex.schema.raw(`
+    alter table campaign rename to all_campaign;
+
+    alter table all_campaign add column is_template boolean not null default false;
+
+    create view campaign as
+      select
+          id
+        , organization_id
+        , title
+        , description
+        , is_started
+        , due_by
+        , created_at
+        , is_archived
+        , use_dynamic_assignment
+        , logo_image_url
+        , intro_html
+        , primary_color
+        , texting_hours_start
+        , texting_hours_end
+        , timezone
+        , creator_id
+        , is_autoassign_enabled
+        , limit_assignment_to_teams
+        , updated_at
+        , replies_stale_after_minutes
+        , landlines_filtered
+        , external_system_id
+        , is_approved
+        , autosend_status
+        , autosend_user_id
+      from all_campaign
+      where is_template = false;
+
+    create or replace view assignable_campaigns as (
+      select id, title, organization_id, limit_assignment_to_teams
+      from campaign
+      where is_started = true
+        and is_archived = false
+        and is_autoassign_enabled = true
+    );
+
+    create or replace view assignable_needs_message as (
+      select acc.id, acc.campaign_id, acc.message_status
+      from assignable_campaign_contacts as acc
+      join campaign on campaign.id = acc.campaign_id
+      where message_status = 'needsMessage'
+        and (
+            ( acc.contact_timezone is null
```

Comment:
Suggestion: Create `assignable_campaign_within_sending_time` or similar, so when new views are going to be created, they can extend that view instead of having to add the within sending time logic.